### PR TITLE
Updated Base Code for adding helper string to designer properties #1128 

### PR DESCRIPTION
**Pull request by suvamjain**
### **Issue - Add helpString to Designer Properties #1128 ** 

Two files have been updated here - 

1. .**./components/annotations/DesignerProperty.java**

DesignerProperty

2. **../components/scripts/ComponentDescriptorGenerator.java**

ComponentDescriptorGenerator

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by suvamjain** [comment]
Sir, can it be merged now or do I need to add anything?

File: appinventor/components/src/com/google/appinventor/components/scripts/ComponentDescriptorGenerator.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
  *   }*,
  *   "properties": [
  *     { "name": "PROPERTY-NAME",
+ *        "description": "DESCRIPTION",  //[suvam, 2019/03/25] Added description field to json
```
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/scripts/ComponentDescriptorGenerator.java]: Please remove the comment within the comment. I think it's clear from the other comment further down when and why this was added. Removing the comment keeps the JSON template clean.

Comment (suvamjain) [reply appinventor/components/src/com/google/appinventor/components/scripts/ComponentDescriptorGenerator.java]: It's done now

**Comment by ewpatton** [comment]
@suvamjain After looking at everything, I think the last thing that we need to do is update ComponentDatabaseInterface.java to read this information in and store it as a field so that we can then pull it out in other code. I have a separate branch I've been working on to add helper icons for the properties, so once you do this I can rebase that branch on top of your work to finish this up.

**Comment by suvamjain** [comment]
@ewpatton I have implemented the changes mentioned above in `ComponentDatabaseInterface.java` file. Please review it and let me know in case of any discrepancies.

**Comment by suvamjain** [comment]
Hello @ewpatton  sir, I have implemented the changes that you mentioned above in the `ComponentDatabaseInterface.java` file. Also I need to update the `ComponentDatabase.java` to use the modified constructor of Interface file. Please review it and let me know in case of any discrepancies. Below are the screenshots of updated code snippets - 

**1. `/src/com/google/appinventor/shared/simple/ComponentDatabaseInterface.java`**

Interface

**2. `/src/com/google/appinventor/client/editor/simple/ComponentDatabase.java`**

Database

**Comment by suvamjain** [comment]
Hi @ewpatton sir, any feedback on the latest PR?

**Comment by ewpatton** [comment]
This change is superseded by #2886

**Pull request closed without merging**